### PR TITLE
Memory Table UI

### LINKED ISSUE #23: Visualize the agent's memory in dashboard
## Type of Issue

Select the type of issue:
- [ ] Bug report (to report a bug)
- [x] Feature request (to request an additional feature)
- [ ] Tracker (I am just using this as a tracker)
- [ ] Documentation Ask

## Description

This feature would add a new React component to the droidlet dashboard that can show the state of the agent's memory with a text based filter. Once this is in, this can evolve into a SQL browser that loads the schema from backend and lets developers query the current memory from the frontend component.

Please feel free to use any open sourced libraries as you see fit.

In the backend, [this](https://github.com/facebookresearch/droidlet/blob/main/base_agent/loco_mc_agent.py#L274) is where the state of the memory is being communicated through a socket event.

To add your component:
1. Add the React component `js` and `css` files [here](https://github.com/facebookresearch/droidlet/tree/main/dldashboard/web/src/components)
2. Add the component to [index.js](https://github.com/facebookresearch/droidlet/blob/main/dldashboard/web/src/index.js)
3. Manage and set state of component [here](https://github.com/facebookresearch/droidlet/blob/main/dldashboard/web/src/StateManager.js)
4. Add new dependencies to [this](https://github.com/facebookresearch/droidlet/blob/main/dldashboard/web/package.json)

## Current Behavior

This is the current state of the dashboard with a 2D grid view of memory : 

Screen Shot 2020-12-04 at 4 30 16 PM

## Expected Behavior

A new pane next to the "Memory 2D" pane that helps visualize the state of memory at backend and eventually allows developers to write SQL queries and query them against the memory from this pane.

## Checklist

Use this checklist if you are using this issue as a tracker:

- [ ] Task 1
- [ ] Task 2
- [ ] Task 3 ...

### PR DESCRIPTION
**Pull request by s0mya**
# Description

WIP This PR will implement a new table view of the Agent's memory.

The completed (and pending) sub tasks are as follows.

- [X] Create a new panel to show the objects in agents memory in a table. 
- [x] Create a new Memory Manager component that will provide an easier to use interface for the table to query it's data.
- [x] Implement nested rows to show the location and abstractions for a memory
- [x] Refactor the Table UI to support virtualized rows for paging and performance.
- [x] Implement a search interface to filter results
- [x] Implement dark theme

Fixes #23 

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [X] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change

## Type of requested review

- [x] I want a thorough review of the implementation.
- [X] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

### Memory Table
image

### Memory Details
image

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [X] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

**Comment by kavyasrinet** [comment]
FYI: Also added @soumith as a reviewer on this

**Review by soumith — DISMISSED** [review]
package-lock.json shouldn't be committed afaik.

**Review by soumith — COMMENTED** [review]
yarn.lock seems to have changed significantly, any idea what's going on? if it's just a bunch of package upgrades, seems okay.

**Comment by s0mya** [comment]
> package-lock.json shouldn't be committed afaik.

Interesting, I was expecting this to be like the other package managers where the lock file is used track of the resolved package versions at the time. But then again I also see the same info yarn.lock. I'll dig around to find out the right approach.

**Comment by soumith** [comment]
so, we are using `yarn`, which for back-compatibility reasons routes through `npm` for packages too. i believe the practice is to commit `yarn.lock` but not other lock files.

**Comment by s0mya** [comment]
> package-lock.json shouldn't be committed afaik.

Fixed with 94643a7

File: dldashboard/web/src/components/Memory/MemoryDetail.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React from "react";
```
Comment (kavyasrinet) [inline dldashboard/web/src/components/Memory/MemoryDetail.js:1]: Can we write component level documentation for these ?
For example why do we need memoryManager , MemoryDetail and MemoryList and MemoryVirtualizedTable and how they serve each other ?

**Review by kavyasrinet — COMMENTED** [review]
Overall implementation looks good but we are lacking documentation. Can we add documentation for the components introduced in this PR ?

File: dldashboard/web/src/components/Memory/MemoryDetail.js (hunk 1 above)
Comment (s0mya) [reply dldashboard/web/src/components/Memory/MemoryDetail.js:1]: @kavyasrinet Yep that makes sense. Do you want me to add comments in the code, or is there something external? LMK if there is a template I should follow. Thanks!

Comment (kavyasrinet) [reply dldashboard/web/src/components/Memory/MemoryDetail.js:1]: Comments in the code or function level doctrings should help.

Comment (s0mya) [reply dldashboard/web/src/components/Memory/MemoryDetail.js:1]: Hi @kavyasrinet, I added the documentation. Let me know if it looks ok?

Comment (kavyasrinet) [reply dldashboard/web/src/components/Memory/MemoryDetail.js:1]: Thanks ! Taking a look now

**Review by kavyasrinet — APPROVED** [review]
Thanks for adding in the documentation. LGTM!

**Comment by s0mya** [comment]
> package-lock.json shouldn't be committed afaik.



> package-lock.json shouldn't be committed afaik.

Fixed with 94643a7

**Pull request merged**